### PR TITLE
[#365] Switch functional DeveloperAppAnalyticsTest: use mock client.

### LINKED ISSUE #365: Switch to mocks for Functional DeveloperAppAnalyticsTest
Switch class to be able use mocks instead of actual calls to Edge org
`tests/src/Functional/DeveloperAppAnalyticsTest.php`

### PR DESCRIPTION
**Pull request by arlina-espinoza**
Refactors the test to use the sdk connector's client, and apigee_mock_api_client.
It reuses some of the mock templates that would be added by #360 , so this PR will need a touch up after that one gets merged.

### CODE REVIEW

**Review by cnovak — APPROVED** [review]
LGTM

File: tests/src/Functional/DeveloperAppAnalyticsTest.php [hunk 1]
```diff
@@ -228,4 +248,36 @@ protected function exportAnalyticsTest() {
...
+
+  /**
+   * Add an app analytics mock response to the stack.
+   *
+   * @param \Drupal\apigee_edge\Entity\DeveloperAppInterface $app
+   *   The app.
+   * @param int $response_code
+   *   Response code, defaults to 200.
+   */
+  protected function queueDeveloperAppResponse(DeveloperAppInterface $app, $response_code = 200) {
```
Comment (cnovak) [inline tests/src/Functional/DeveloperAppAnalyticsTest.php]: Should this function be in `ApigeeMockApiClientHelperTrait` instead for other tests to use?

Comment (shadcn) [reply tests/src/Functional/DeveloperAppAnalyticsTest.php]: @arlina-espinoza Let's move it to the trait. I have a few tests that could use this.

**Comment by shadcn** [comment]
@googlebot I consent.

**Pull request merged**